packages/front-end/components/Experiment/ExperimentGraph.tsx

```diff
@@ -79,60 +90,89 @@ export default function ExperimentGraph({
-                width={xMax + barWidth / 2}
-              />
-              {graphData.map((d, i) => {
-                const barX = xScale(getValidDate(d.name)) - barWidth / 2;
-                let barHeight = yMax - (yScale(d.numExp) ?? 0);
-                // if there are no experiments this month, show a little nub for design reasons.
-                if (barHeight === 0) barHeight = 6;
-                const barY = yMax - barHeight;
-                return (
-                  <BarRounded
-                    key={d.name + i}
-                    x={barX + 5}
-                    y={barY}
-                    width={Math.max(10, barWidth - 10)}
-                    height={barHeight}
-                    fill="#73D1F0"
-                    top
-                    radius={6}
-                  />
-                );
-              })}
+          <>
+            <div style={{ position: "relative" }}>
+              {tooltipOpen && (
+                <TooltipWithBounds
+                  left={tooltipLeft}
+                  top={tooltipTop}
+                  className={styles.tooltip}
+                  unstyled={true}
+                >
+                  {tooltipData} ex
+                </TooltipWithBounds>
+              )}
+            </div>
+            <svg width={width} height={height}>
+              <Group left={margin[3]} top={margin[0]}>
+                <GridRows
+                  scale={yScale}
+                  numTicks={Math.min(maxYValue, 5)}
+                  width={xMax + barWidth / 2}
+                />
+                {graphData.map((d, i) => {
+                  const barX = xScale(getValidDate(d.name)) - barWidth / 2;
+                  let barHeight = yMax - (yScale(d.numExp) ?? 0);
+                  // if there are no experiments this month, show a little nub for design reasons.
+                  if (barHeight === 0) barHeight = 6;
+                  const barY = yMax - barHeight;
+                  return (
+                    <BarRounded
+                      key={d.name + i}
+                      x={barX + 5}
+                      y={barY}
+                      width={Math.max(10, barWidth - 10)}
+                      height={barHeight}
+                      fill={"#73D1F0"}
+                      top
+                      radius={6}
+                      className={styles.barHov}
+                      onMouseMove={() => {
+                        const top = barY - 25;
+                        const left = barX + Math.max(10, barWidth - 10);
+                        showTooltip({
+                          tooltipTop: top,
+                          tooltipLeft: left,
+                          tooltipData: d.numExp,
+                        });
+                      }}
+                      onMouseLeave={() => {
+                        window.setTimeout(() => {
+                          hideTooltip();
+                        }, 100);
+                      }}
```

Comment:
I think this is fine for now.  I could see maybe moving the mouse events up to the svg element instead of having them on each bar.  The bars are really thin, so it can be hard to hover over them directly.  If you are watching mouseMove on the svg element instead you can expand the hit area and show the tooltip whenever they get close to the bar.

Not a big deal though.  I'm fine merging it now, unless you feel like restructuring the code.

Reply:
@jdorn Sounds good, thanks! I'll work on restructuring the code. Might take me a couple of days though. I'll defer to you on whether to merge now.

Reply:
Hey @jdorn, I'm currently stuck. It would be great if you could point me towards the right direction, but I understand if you can't. I could create an issue, and let someone else restructure the code. 

I've tried going different routes to solve the problem. The closest solution I've come up with involves using `localPoint` from `@visx/event `. Something like: 

```
const handlePointer = (event) => {
          const coords = localPoint(event);
          showTooltip({
               tooltipTop: coords.y,
               tooltipLeft: coords.x,
               tooltipData: 'test',
            });
        }
<svg onMouseOver={handlePointer}/>
```

Problem with this is that it will all depend on the mouse position at the time of hovering over the bars. Resulting in different positions for the toolTip. 

Let me know if I should create an issue, or try a different approach. Thanks!

Reply:
I think you are close.  You'll need to use `onMouseMove` instead of `onMouseOver` for the svg so it continues to fire as you move around the graph.  You can use the x coordinate of the mouse to figure out the closest bar in the graph.  Then, you can use that bar to set the tooltip top, left, and data.  Does that make sense?

Reply:
Oh yeah that makes sense. Thanks!